File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Heart Rate Bricklet

	// Callback function for heart rate callback
	static void HeartRateCB(BrickletHeartRate sender, int heartRate)
	{
		Console.WriteLine("Heart Rate: " + heartRate + " bpm");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletHeartRate hr = new BrickletHeartRate(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register heart rate callback to function HeartRateCB
		hr.HeartRateCallback += HeartRateCB;

		// Set period for heart rate callback to 1s (1000ms)
		// Note: The heart rate callback is only called every second
		//       if the heart rate has changed since the last call!
		hr.SetHeartRateCallbackPeriod(1000);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_callback.rs
use std::{error::Error, io, thread};
use tinkerforge::{heart_rate_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Heart Rate Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let hr = HeartRateBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    let heart_rate_receiver = hr.get_heart_rate_callback_receiver();

    // Spawn thread to handle received callback messages.
    // This thread ends when the `hr` object
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for heart_rate in heart_rate_receiver {
            println!("Heart Rate: {} bpm", heart_rate);
        }
    });

    // Set period for heart rate receiver to 1s (1000ms).
    // Note: The heart rate callback is only called every second
    //       if the heart rate has changed since the last call!
    hr.set_heart_rate_callback_period(1000);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
